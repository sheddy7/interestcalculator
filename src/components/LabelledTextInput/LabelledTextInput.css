.labelled-row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  margin-bottom: 10px;
}

.label-text {
  width: 140px;
  padding-left: 10px;
}