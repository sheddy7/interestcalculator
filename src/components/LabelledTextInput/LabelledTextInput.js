import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../../components/TextInput/TextInput';
import './LabelledTextInput.css';

const LabelledTextInput = ({ label, postLabel, onChange, value }) => {
  return (
    <div className="labelled-row">
      <p className="label-text">{label}</p>
      <TextInput onChange={onChange} value={value} />
      <p className="label-text">{postLabel}</p>
    </div>
  );
};

LabelledTextInput.propTypes = {
  label: PropTypes.string.isRequired,
  postLabel: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.number
};

LabelledTextInput.defaultProps = {
  postLabel: null,
  value: 0
};

export default LabelledTextInput;
