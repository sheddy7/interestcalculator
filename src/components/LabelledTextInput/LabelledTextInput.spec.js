import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LabelledTextInput from './LabelledTextInput';

configure({ adapter: new Adapter() });

describe(LabelledTextInput.name, () => {
  describe('given default props', () => {
    const onChange = jest.fn();
    let wrapper;

    it('renders successfully', () => {
      wrapper = shallow(
        <LabelledTextInput label="amount" onChange={onChange} />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('triggering onChange function passed to TextInput fires onChange prop', () => {
      wrapper
        .find('WithStyles(TextInput)')
        .props()
        .onChange();
      expect(onChange).toHaveBeenCalledTimes(1);
    });
  });
});
