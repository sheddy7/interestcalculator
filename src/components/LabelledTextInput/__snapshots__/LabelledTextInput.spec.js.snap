// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledTextInput given default props renders successfully 1`] = `
<div
  className="labelled-row"
>
  <p
    className="label-text"
  >
    amount
  </p>
  <WithStyles(TextInput)
    onChange={[MockFunction]}
    value={0}
  />
  <p
    className="label-text"
  />
</div>
`;
